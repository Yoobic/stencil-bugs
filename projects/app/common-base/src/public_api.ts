export { AppCommonBaseModule } from './lib/app-common-base.module';

export * from './lib/bases/app-chat-page/app-chat-page.component';
export * from './lib/bases/app-entity-action-page/app-entity-action-page.component';
export * from './lib/bases/app-form-dynamic-page/app-form-dynamic-page.component';
export * from './lib/bases/app-grid-dialog-page/app-grid-dialog-page.component';
export * from './lib/bases/app-menu-page/app-menu-page.component';
export * from './lib/bases/app-root/app-root.component';
export * from './lib/bases/app-search-page/app-search-page.component';
export * from './lib/bases/app-walkthrough-page/app-walkthrough-page.component';
export * from './lib/bases/app-whatsnew-page/app-whatsnew-page.component';
export * from './lib/bases/base-page/base-page.component';

export * from './lib/components/app-profile/app-profile.component';
export * from './lib/components/app-login/app-login.component';
export * from './lib/components/app-grid/app-grid.component';
export * from './lib/components/app-form-dynamic/app-form-dynamic.component';

export * from './lib/services/dialog/dialog.service';
export * from './lib/services/utils/utils.service';

export * from './lib/animations/animations';
export * from './lib/animations/transitions';

export * from './lib/modals/app-result-dialog-page/app-result-dialog-page.component';