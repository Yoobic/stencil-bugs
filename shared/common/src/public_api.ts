export { CommonModule } from './lib/common.module';

export * from './lib/decorators/mixin/mixin.decorator';
export * from './lib/decorators/persistent/persistent.decorator';
export * from './lib/directives/long-press/long-press.directive';

export * from './lib/interfaces/animate-state/animate-state.class';
export * from './lib/interfaces/animations/animations.class';
export * from './lib/interfaces/colors/colors.interface';
export * from './lib/interfaces/config-constants/config-constants.interface';
export * from './lib/interfaces/position/position.interface';

export * from './lib/pipes/array-from-number/array-from-number.pipe';
export * from './lib/pipes/cloudinary/cloudinary.pipe';
export * from './lib/pipes/date-format/date-format.pipe';
export * from './lib/pipes/file-size/file-size.pipe';
export * from './lib/pipes/highlight/highlight.pipe';
export * from './lib/pipes/https/https.pipe';
export * from './lib/pipes/keys/keys.pipe';
export * from './lib/pipes/limit-to/limit-to.pipe';
export * from './lib/pipes/map-values/map-values.pipe';
export * from './lib/pipes/newline/newline.pipe';
export * from './lib/pipes/nohtml/nohtml.pipe';
export * from './lib/pipes/order-by/order-by.pipe';
export * from './lib/pipes/round/round.pipe';
export * from './lib/pipes/safe/safe.pipe';
export * from './lib/pipes/take/take.pipe';
export * from './lib/pipes/time-ago/time-ago.pipe';
export * from './lib/pipes/timer/timer.pipe';

export * from './lib/services/config-constants/config-constants.service';
export * from './lib/services/core-config/core-config.service';
export * from './lib/services/dom-adapter/dom-adapter.service';
export * from './lib/services/exif-restorer/exif-restorer.service';
export * from './lib/services/geo-location/geo-location.service';
export * from './lib/services/local-forage/local-forage.service';
export * from './lib/services/local-storage/local-storage.service';
export * from './lib/services/log/log.service';
export * from './lib/services/network/network.service';
export * from './lib/services/promise/promise.service';
export * from './lib/services/resize/resize.service';
export * from './lib/services/ruler/ruler.service';
export * from './lib/services/utils/utils.service';

export * from './lib/mixins/lodash-mixins/lodash-mixins';