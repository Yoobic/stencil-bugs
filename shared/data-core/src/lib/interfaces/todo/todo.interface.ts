import { Model } from '../../decorators/model/model.decorator';
import { Editable } from '../../decorators/editable/editable.decorator';
import { IFormField, FormFieldType, ITodo, ITodoTask, ITodoTaskSimple } from '@shared/interfaces';
import { User } from '../user/user.interface';

export function onUserMeActionHandler() {
    this.selectMatch(this.session.user);
}

@Model({ className: 'Todo' })
export class Todo extends ITodo {

    @Editable('Todo', { type: FormFieldType.text, name: 'title', required: true })
    title: string;

    @Editable('Todo', {
        title: 'ASSIGNTO', name: 'user', type: FormFieldType.autocomplete, collectionName: 'user',
        queryFields: User.getSimpleFields(), multiple: false, clearable: true, required: true,
        extraButton: { title: 'ASSIGNTO', buttons: [{ text: 'ME', handler: onUserMeActionHandler }] }
    })
    user: any;

    @Editable('Todo', { title: 'DUEDATE', type: FormFieldType.date, required: false, clearable: true, minDate: new Date() })
    duedate: Date;

    @Editable('Todo', { title: 'NOTIFICATIONEMAILS', type: FormFieldType.emailreport, showUsers: true, tab: 'INFORMATION', exportOrder: 11, advanced: true })
    notificationemail?: Array<string>;

    @Editable('Todo', { title: 'REMINDER', type: FormFieldType.datetime, required: false, clearable: true, advanced: true })
    reminderdate?: Date;

    @Editable('Todo', { type: FormFieldType.starrating, advanced: true })
    priority?: string;

    field?: IFormField;
}

@Model({ className: 'TodoTask' })
export class TodoTask extends ITodoTask {
    @Editable('TodoTask', { type: FormFieldType.text, title: 'TITLE', required: true })
    text: { value: string };

    @Editable('TodoTask', { type: FormFieldType.textarea, required: false })
    comments?: { value: string };

    @Editable('TodoTask', { type: FormFieldType.date, required: false, clearable: true, minDate: new Date() })
    duedate?: { value: Date };

    @Editable('TodoTask', { type: FormFieldType.toggle, required: false, title: 'PHOTO', flex: 50 })
    hasphoto?: { value: boolean };

    @Editable('TodoTask', { type: FormFieldType.toggle, required: false, title: 'MANDATORYPHOTO', flex: 50, condition: 'hasphoto.value==1' })
    isphotorequired?: { value: boolean };

    @Editable('TodoTask', { type: FormFieldType.toggle, required: false, title: 'ALLOWLIBRARY', flex: 50, condition: 'hasphoto.value==1' })
    allowLibrary?: { value: boolean };

    field?: IFormField;
}

@Model({ className: 'TodoTaskSimple' })
export class TodoTaskSimple extends ITodoTaskSimple {
    @Editable('TodoTaskSimple', { type: FormFieldType.textarea, title: 'DESCRIPTION', required: true })
    text: { value: string };

    @Editable('TodoTaskSimple', {
        title: 'ASSIGNTO', name: 'user', type: FormFieldType.autocomplete,
        queryFields: User.getSimpleFields(),
        collectionName: 'user', multiple: false, clearable: true, required: true,
        extraButton: { title: 'ASSIGNTO', buttons: [{ text: 'ME', handler: onUserMeActionHandler }] }
    })
    user: any;

    @Editable('TodoTaskSimple', { type: FormFieldType.date, required: false, clearable: true, minDate: new Date() })
    duedate?: { value: Date };

    @Editable('TodoTaskSimple', { type: FormFieldType.toggle, required: false, title: 'ASKFORPHOTO', flex: 50 })
    hasphoto?: { value: boolean };

    @Editable('TodoTaskSimple', { type: FormFieldType.toggle, required: false, title: 'MANDATORYPHOTO', flex: 50, condition: 'hasphoto.value==1' })
    isphotorequired?: { value: boolean };

    @Editable('TodoTaskSimple', { type: FormFieldType.toggle, required: false, title: 'ALLOWLIBRARY', flex: 50, condition: 'hasphoto.value==1' })
    allowLibrary?: { value: boolean };
    field?: IFormField;

}
