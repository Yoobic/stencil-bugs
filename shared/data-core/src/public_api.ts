export { DataCoreModule } from './lib/data-core.module';

export * from './lib/decorators/editable/editable.decorator';
export * from './lib/decorators/mapping/mapping.decorator';
export * from './lib/decorators/model/model.decorator';
export * from './lib/decorators/searchable/searchable.decorator';

export * from './lib/interfaces/aggregate-log/aggregate-log.interface';
export * from './lib/interfaces/algorithm/algorithm.interface';
export * from './lib/interfaces/autorenew-config/autorenew-config.interface';
export * from './lib/interfaces/backup/backup.interface';
export * from './lib/interfaces/catalog/catalog.interface';
export * from './lib/interfaces/chart-definition/chart-definition.interface';
export * from './lib/interfaces/condition/condition.interface';
export * from './lib/interfaces/condition/icondition.interface';
export * from './lib/interfaces/constants/constants.interface';
export * from './lib/interfaces/currency/currency.interface';
export * from './lib/interfaces/dashboard/dashboard.interface';
export * from './lib/interfaces/entity/entity.interface';
export * from './lib/interfaces/field-selector/field-selector.interface';
export * from './lib/interfaces/file/file.interface';
export * from './lib/interfaces/folder/folder.interface';
export * from './lib/interfaces/group/group.interface';
export * from './lib/interfaces/kpi/kpi.interface';
export * from './lib/interfaces/location-type/location-type.interface';
export * from './lib/interfaces/location/location.interface';
export * from './lib/interfaces/mapping/mapping.interface';
export * from './lib/interfaces/mission-description/mission-description.interface';
export * from './lib/interfaces/model/model.interface';
export * from './lib/interfaces/notification-event/notification-event.interface';
export * from './lib/interfaces/notification/notification.interface';
export * from './lib/interfaces/operation/operation.interface';
export * from './lib/interfaces/product/product.interface';
export * from './lib/interfaces/response-object/response-object.interface';
export * from './lib/interfaces/slide/slide.interface';
export * from './lib/interfaces/sort/sort.interface';
export * from './lib/interfaces/tableau/tableau.interface';
export * from './lib/interfaces/tag-group/tag-group.interface';
export * from './lib/interfaces/tag/tag.interface';
export * from './lib/interfaces/tenant/tenant.interface';
export * from './lib/interfaces/todo/todo.interface';
export * from './lib/interfaces/translation/translation.interface';
export * from './lib/interfaces/user/user.interface';
export * from './lib/interfaces/workplace/workplace.interface';

export * from './lib/pipes/authorized/authorized.pipe';
export * from './lib/pipes/currency/currency.pipe';
export * from './lib/pipes/filter/filter.pipe';
export * from './lib/pipes/username/username.pipe';
export * from './lib/pipes/useremail/useremail.pipe';
export * from './lib/pipes/userinitial/userinitial.pipe';

export * from './lib/services/activity/activity.service';
export * from './lib/services/algorithms/algorithms.service';
export * from './lib/services/authentication/authentication.guard';
export * from './lib/services/authentication/authentication.service';
export * from './lib/services/authentication/current-session.resolver';
export * from './lib/services/backup/backup.service';
export * from './lib/services/box/box.service';
export * from './lib/services/blog/blog.service';
export * from './lib/services/links/links.service';
export * from './lib/services/broker/broker.service';
export * from './lib/services/config/config.service';
export { Dashboard as DashboardCore } from './lib/services/dashboard/dashboard.service';
export * from './lib/services/data-loader/data-loader.service';
export * from './lib/services/files/files.service';
export * from './lib/services/files/files-broker.service';
export * from './lib/services/googlemaps/googlemaps.service';
export * from './lib/services/googletranslate/googletranslate.service';
export * from './lib/services/http-interceptor/http-interceptor.service';
export * from './lib/services/loading-bar/loading-bar.service';
export * from './lib/services/locations/locations.service';
export * from './lib/services/loopback/loopback.service';
export * from './lib/services/models/models.service';
export * from './lib/services/print/print.service';
export * from './lib/services/push/push.service';
export * from './lib/services/requestor/requestor.service';
export * from './lib/services/security/security.service';
export * from './lib/services/session/session.service';
export * from './lib/services/smartloc/smartloc.service';
export * from './lib/services/users/users.service';
export * from './lib/services/version/version.service';
export * from './lib/services/workplace/workplace.service';
export * from './lib/services/xlsx/xlsx.service';

export * from './lib/validators/password-validator/password-validator';
export * from './lib/validators/username-validator/username-validator';
