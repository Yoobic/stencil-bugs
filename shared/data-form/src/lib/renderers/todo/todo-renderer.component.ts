import { Component } from '@angular/core';
import { ICellRendererAngularComp } from 'ag-grid-angular';
import { DefaultRendererComponent } from '../default/default-renderer.component';

import { map as _map } from 'lodash-es';

@Component({
    selector: 'todo-cell',
    template: `{{value()}}`
})
export class TodoRendererComponent implements ICellRendererAngularComp {
    private params: any;

    agInit(params: any) {
        this.params = params;
    }

    value() {
        let value = DefaultRendererComponent.renderer(this.params);
        if (value && value.values) {
            return _map(value.values, (v: any) => v.text.value).join(', ');
        }
        return '';
    }

    refresh(params: any) {
        return false;
    }
}