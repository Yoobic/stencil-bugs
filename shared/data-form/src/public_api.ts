
export { DataFormModule } from './lib/data-form.module';

export * from './lib/interfaces/custom-form-field/custom-form-field.interface';
export * from './lib/interfaces/form-field/form-field.class';
export * from './lib/interfaces/gridstate/gridstate.interface';
export * from './lib/interfaces/cache/cache.interface';
export * from './lib/interfaces/custom-model/custom-model.interface';

export * from './lib/services/cell-renderer/cell-renderer.service';
export * from './lib/services/chart/chart.service';
export * from './lib/services/form-dynamic-builder/form-dynamic-builder.service';
export * from './lib/services/conditions/conditions.service';
export * from './lib/services/mapping/mapping.service';
export * from './lib/services/custom-models/custom-models.service';

export * from './lib/validators/form-filter-field-validator/form-filter-field-validator';
export * from './lib/validators/form-input-base-email-validator/form-input-base-email-validator';
export * from './lib/validators/form-input-base-number-validator/form-input-base-number-validator';
export * from './lib/validators/form-input-between-validator/form-input-between-validator';
export * from './lib/validators/form-input-checklist-validator/form-input-checklist-validator';
export * from './lib/validators/form-input-missionfield-validator/form-input-missionfield-validator';
export * from './lib/validators/form-input-multiphotos-validator/form-input-multiphotos-validator';
export * from './lib/validators/form-input-photo-validator/form-input-photo-validator';
export * from './lib/validators/form-input-select-validator/form-input-select-validator';
export * from './lib/validators/password-are-not-matching-validator/password-are-not-matching-validator';

export * from './lib/renderers/address/address-renderer.component';
export * from './lib/renderers/boolean/boolean-renderer.component';
export * from './lib/renderers/button/button-renderer.component';
export * from './lib/renderers/checklist/checklist-renderer.component';
export * from './lib/renderers/conformity-progress/conformity-progress-renderer.component';
export * from './lib/renderers/conformity-relative-progress/conformity-relative-progress-renderer.component';
export * from './lib/renderers/custom-model/custom-model-renderer.component';
export * from './lib/renderers/datetime/datetime-renderer.component';
export * from './lib/renderers/default/default-renderer.component';
export * from './lib/renderers/delta/delta-renderer.component';
export * from './lib/renderers/distance/distance-renderer.component';
export * from './lib/renderers/duration/duration-renderer.component';
export * from './lib/renderers/location/location-renderer.component';
export * from './lib/renderers/mission-status/mission-status-renderer.component';
export * from './lib/renderers/mission-validation/mission-validation-renderer.component';
export * from './lib/renderers/number/number-renderer.component';
export * from './lib/renderers/multiphotos/multiphotos-renderer.component';
export * from './lib/renderers/percentage/percentage-renderer.component';
export * from './lib/renderers/photo/photo-renderer.component';
export * from './lib/renderers/progress/progress-renderer.component';
export * from './lib/renderers/time/time-renderer.component';
export * from './lib/renderers/timer/timer-renderer.component';
export * from './lib/renderers/todo/todo-renderer.component';
export * from './lib/renderers/user-full-name/user-full-name-renderer.component';
export * from './lib/renderers/validation-progress/validation-progress-renderer.component';
export * from './lib/renderers/video/video-renderer.component';