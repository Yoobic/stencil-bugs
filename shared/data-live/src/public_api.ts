
export { DataLiveModule } from './lib/data-live.module';

export { Channel as ChannelInterface } from './lib/interfaces/channel/channel.interface';
export * from './lib/interfaces/channels/channels.interface';
export * from './lib/interfaces/message/message.interface';
export * from './lib/interfaces/message/message.model';

export * from './lib/services/channel/channel.service';
export * from './lib/services/intercom/intercom.service';
export * from './lib/services/messages/messages.service';
export * from './lib/services/pubnub/pubnub.service';
export * from './lib/services/track/track.service';
export * from './lib/services/twilio-token/twilio-token.service';
export * from './lib/services/raven-error-handler/raven-error-handler.service';