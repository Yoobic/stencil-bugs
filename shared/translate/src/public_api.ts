export { TranslateModule } from './lib/translate.module';

export * from './lib/pipes/distance/distance.pipe';
export * from './lib/pipes/filter/filter.pipe';
export * from './lib/pipes/polyglot/polyglot.pipe';

export * from './lib/services/translate/translate.service';