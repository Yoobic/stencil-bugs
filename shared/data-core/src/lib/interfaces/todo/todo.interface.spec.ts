import { TodoTask, TodoTaskSimple } from './todo.interface';
import { Models } from '../../../public_api';

describe('shared: data-core: interfaces: TodoTask', () => {
    it('should create a todo task entity', () => {
        let entity = new TodoTask();
        expect(entity._id).toBeUndefined();
        let model = Models.getModel('TodoTask');
        expect(model).toBeTruthy();
        let modelCopy = {...model};
        modelCopy['_formFields'][2].minDate = '2000-01-01T00:00:00.000Z';
        expect(modelCopy).toMatchSnapshot();
    });
    it('should create a TodoTaskSimple entity', () => {
        let entity = new TodoTaskSimple();
        expect(entity._id).toBeUndefined();
        let model = Models.getModel('TodoTaskSimple');
        expect(model).toBeTruthy();
        let modelCopy = {...model};
        modelCopy['_formFields'][2].minDate = '2000-01-01T00:00:00.000Z';
        expect(modelCopy).toMatchSnapshot();
    });
});
