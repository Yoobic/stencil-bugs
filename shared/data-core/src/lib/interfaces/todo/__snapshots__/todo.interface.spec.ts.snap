// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shared: data-core: interfaces: TodoTask should create a TodoTaskSimple entity 1`] = `
Object {
  "_formFields": Array [
    Object {
      "filterName": "text",
      "name": "text",
      "required": true,
      "title": "DESCRIPTION",
      "type": "textarea",
    },
    Object {
      "clearable": true,
      "collectionName": "user",
      "extraButton": Object {
        "buttons": Array [
          Object {
            "handler": [Function],
            "text": "ME",
          },
        ],
        "title": "ASSIGNTO",
      },
      "filterName": "userRef",
      "multiple": false,
      "name": "user",
      "queryFields": Array [
        "_id",
        "imageData",
        "firstName",
        "lastName",
        "email",
        "telephone",
        "username",
      ],
      "required": true,
      "title": "ASSIGNTO",
      "type": "autocomplete",
    },
    Object {
      "clearable": true,
      "filterName": "duedate",
      "minDate": "2000-01-01T00:00:00.000Z",
      "name": "duedate",
      "required": false,
      "title": "DUEDATE",
      "type": "date",
    },
    Object {
      "filterName": "hasphoto",
      "flex": 50,
      "name": "hasphoto",
      "required": false,
      "title": "ASKFORPHOTO",
      "type": "toggle",
    },
    Object {
      "condition": "hasphoto.value==1",
      "filterName": "isphotorequired",
      "flex": 50,
      "name": "isphotorequired",
      "required": false,
      "title": "MANDATORYPHOTO",
      "type": "toggle",
    },
    Object {
      "condition": "hasphoto.value==1",
      "filterName": "allowLibrary",
      "flex": 50,
      "name": "allowLibrary",
      "required": false,
      "title": "ALLOWLIBRARY",
      "type": "toggle",
    },
  ],
  "appareance": Map {},
  "className": "TodoTaskSimple",
  "collectionName": undefined,
  "feathersService": undefined,
  "fields": undefined,
  "include": undefined,
  "isCustom": undefined,
  "mappingFields": Map {},
  "searchSubquery": undefined,
  "searchableFields": Array [],
  "type": [Function],
}
`;

exports[`shared: data-core: interfaces: TodoTask should create a todo task entity 1`] = `
Object {
  "_formFields": Array [
    Object {
      "filterName": "text",
      "name": "text",
      "required": true,
      "title": "TITLE",
      "type": "text",
    },
    Object {
      "filterName": "comments",
      "name": "comments",
      "required": false,
      "title": "COMMENTS",
      "type": "textarea",
    },
    Object {
      "clearable": true,
      "filterName": "duedate",
      "minDate": "2000-01-01T00:00:00.000Z",
      "name": "duedate",
      "required": false,
      "title": "DUEDATE",
      "type": "date",
    },
    Object {
      "filterName": "hasphoto",
      "flex": 50,
      "name": "hasphoto",
      "required": false,
      "title": "PHOTO",
      "type": "toggle",
    },
    Object {
      "condition": "hasphoto.value==1",
      "filterName": "isphotorequired",
      "flex": 50,
      "name": "isphotorequired",
      "required": false,
      "title": "MANDATORYPHOTO",
      "type": "toggle",
    },
    Object {
      "condition": "hasphoto.value==1",
      "filterName": "allowLibrary",
      "flex": 50,
      "name": "allowLibrary",
      "required": false,
      "title": "ALLOWLIBRARY",
      "type": "toggle",
    },
  ],
  "appareance": Map {},
  "className": "TodoTask",
  "collectionName": undefined,
  "feathersService": undefined,
  "fields": undefined,
  "include": undefined,
  "isCustom": undefined,
  "mappingFields": Map {},
  "searchSubquery": undefined,
  "searchableFields": Array [],
  "type": [Function],
}
`;
